src/synthesis.ts

```diff
@@ -1,38 +1,56 @@
 import {
   availableAmount,
   Effect,
+  mallPrice,
   mySpleenUse,
   retrieveItem,
   spleenLimit,
   sweetSynthesis,
   sweetSynthesisResult,
 } from "kolmafia";
-import { $items, clamp } from "libram";
+import { $item, $items, clamp } from "libram";
 import { shuffle } from "./lib";
 
-const whitelist = $items`sugar shotgun, sugar shillelagh, sugar shank, sugar chapeau, sugar shorts, sugar shield, sugar shirt, Fudgie Roll`;
+const allowList = $items`Fudgie Roll, peanut brittle shield, sugar shotgun, sugar shillelagh, sugar shank, sugar chapeau, sugar shorts, sugar shield, sugar shirt`;
+
+// For safety, explicitly skip candies that are no longer obtainable or extremely rare
+const blockList = new Set([
+  $item`candied nuts`,
+  $item`candy kneecapping stick`,
+  $item`chocolate cigar`,
+  $item`fancy but probably evil chocolate`,
+  $item`fancy chocolate`,
+  $item`fancy chocolate car`,
+  $item`gummi ammonite`,
+  $item`gummi belemnite`,
+  $item`gummi trilobite`,
+  $item`powdered candy sushi set`,
+  $item`radio button candy`,
+  $item`spiritual candy cane`,
+  $item`Ultra Mega Sour Ball`,
+  $item`vitachoconutriment capsule`,
+]);
+
+const buyable = $items
+  .all()
+  .filter((i) => i.tradeable && i.candyType === "complex" && !blockList.has(i))
+  .sort((a, b) => mallPrice(a) - mallPrice(b))
```

Comment:
this will do a very large number of mallsearches at the start of operations regardless of whether a player owns synth or has sufficient organ space to cast it

Reply:
In the interim I'll move it inside the function.